src/packages/cli/src/cli.ts

```diff
@@ -31,13 +32,19 @@ const logAndForceExit = (messages: any[], exitCode = 0) => {
 const version = process.env.VERSION || "DEV";
 const cliVersion = process.env.CLI_VERSION || "DEV";
 const coreVersion = process.env.CORE_VERSION || "DEV";
+let versionCheck = new VersionCheck(version);
```

Comment:
`setPackageName`
https://github.com/trufflesuite/ganache/pull/3285/files/0644054b8458eafdb52f2a9699842ed0c91f6a4e#diff-a04d5e4634c93d8c1106b8a07fbcef64bc6a2258986fba0e6c504541e6a54f2bR108

I don't think we should overload the `packageName` that is sent to the server to retrieve the version should also be used to determine placement on the file system.

If we want to support a dynamic location `Conf` details how to accomplish that and we should have a different setting/conf/whatever recommendation that affects that.

Reply:
Update 2:

`Conf` will create a directory based on the dependency

`/home/user/.config/@ganache/version-check-nodejs/config.json`

If that is a problem. If we change the package name for `Conf` I believe it would just be 

`/home/user/.config/@ganache/ganache-cli/config.json`

Reply:
In a _bundled_ `ganache`, the one we ship to npm, `Conf` will create it here by default:

`/home/user/.config/ganache-nodejs/config.json`

I really don't like the `"-nodejs"` suffix, and we already avoid it in Ganache UI, persistent-cache, and now detach mode. I think we should do the same in version check by setting Confi's `projectSuffix` option to `""`. Note: by dropping `"-nodejs"` we do risk colliding with other tools named `ganache`, but I think that is extremely unlikely, and besides, we were there first! :-p

If we do drop the suffix it gets kinda funny, we will now clobber with ourselves, as Ganache UI uses `<config directory>/Ganache/` which on Windows is the same thing as `<config directory>/ganache`.

The good news is that we know and control all the files/folders Ganache UI will put there, so we _can_ coexist if we wanted.

We could do something like (not hard coded, the options should come from `new VersionCheck`):

`new Conf({ projectName: "Ganache/version-check", suffix: "" });`

As a bonus, if we give `Conf` a `projectName` option it will skip a couple of start up filesystem reads! It will no longer search up the directory tree for the nearest package.json, and it will skip reading the project `name` from it. See: https://github.com/sindresorhus/conf/blob/92320d7e2c9db2c56b05dd654c3a248a635598d2/source/index.ts#L78.

So, I propose having a _conversation_ about effectively doing: `new Conf({ projectName: "Ganache/version-check",  "suffix": ""});`

Downside: there are lots of Ganache UI files and it will be ugly and messy. It shouldn't be too hard to add migration in UI to make the directory make sense again.

Reply:
If we want to store other config data (like _opt-in_ telemetry settings?) in a global location for Ganache in the future it'd be really great if we had a single `Conf` instance to manage it all. I've been thinking it may work well for `VersionCheck` to take a `Conf` as an argument, and we could set the Conf `projectName` to `"Ganache/cli"` (and `suffix` to `""`).

Reply:
We want config files written to `.config/Ganache/cli` We will have a separate `.config/Ganache/ui` 

The name of the file that gets written does not matter but we want the name to make sense if we extend it later.